File: server/StoreDesk.API/Models/Item.cs
namespace StoreDesk.API.Models;

public class Item
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public int Quantity { get; set; }
    public int CategoryId { get; set; }
    public Category? Category { get; set; }
    public ICollection<Transaction> Transactions
    { get; set; } = new List<Transaction>();
}

File: client/src/types/item.ts
export interface Item {
  id: number;
  name: string;
  description?: string;
  price: number;
  quantity: number;
  categoryId: number;
  categoryName?: string;
}

export interface CreateItemPayload {
  name: string;
  description?: string;
  price: number;
  quantity: number;
  categoryId: number;
}
